sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_response_handlers_async.py

```diff
@@ -0,0 +1,42 @@
+import functools
+from urllib.parse import urlparse, parse_qsl
+from azure.core.async_paging import AsyncItemPaged
+from ._response_handlers import healthcare_result, analyze_result
+
+async def healthcare_extract_page_data_async(response, obj, response_headers, health_job_state):
```

Comment:
i think a main reason for the async issues you're running into is because the callbacks you are adding are asynchronous when they shouldn't be. A rule of thumb is: "if you don't have to call `await` on anything in the function, the function shouldn't be labeled as async"